manifests/plugin.pp

```diff
@@ -56,6 +59,7 @@
   Boolean $enabled                  = true,
   String $digest_type               = 'sha1',
   Boolean $pin                      = false,
+  Array $download_options           = ['--http1.1'],
```

Comment:
Should this really be the default?

Reply:
Actually the error is depending on which server behind update.jenins.io is serving the request. I think there is no downside for now to always use http1.1 instead of 2.0?

Reply:
HTTP 2.0 can be more efficient. Have you at least filed an issue with Jenkins to track it? IMHO such a workaround always needs an issue and link it in a code comment.

Reply:
No, there are a few servers behind and one of them is randomly having issues. i still haven't found out which server.
For me it will also be fine to remove the default but make it injectable via parameter so i can override it.

Reply:
Ok. Got one...

`/usr/bin/curl --head https://updates.jenkins.io/latest/ansicolor.hpi -fsSLg --max-redirs 5; echo $?`

```
HTTP/2 302
date: Thu, 28 Jul 2022 12:26:52 GMT
server: Apache
location: https://updates.jenkins.io/download/plugins/ansicolor/1.0.2/ansicolor.hpi
content-type: text/html; charset=iso-8859-1

curl: (16) Error in the HTTP2 framing layer
16
```

Reply:
https://github.com/jenkins-infra/helpdesk/issues/3078

Reply:
Well ... https://github.com/jenkins-infra/helpdesk/issues/3078#issuecomment-1198084336

Reply:
Ok, that certainly helps in giving me assurances that this is the right thing to do.

Reply:
> And why the heck are the tests failing since ages?

It has been very hard to keep them green, mostly because we don't have proper dependency resolution. Ideally we'd have something that utilizes https://github.com/jenkinsci/plugin-installation-manager-tool/ so you don't need to do that yourself.

Reply:
As i can see in the documentation, the tool also won't resolve dependencies for you :(